screen.go

```diff
@@ -250,14 +255,21 @@ func (s *Screen) currentLineForWriting() *screenLine {
 		}
 		s.LinesScrolledOut += scrollOutTo
 
-		// Make a new line on the bottom using a recycled node slice. There's
-		// at least one we just added.
-		r1 := len(s.nodeRecycling) - 1
+		var nodes []node
+		if r1 := len(s.nodeRecycling) - 1; r1 >= 0 {
+			// Make a new line on the bottom using a recycled node slice. There's
+			// usually at least one we just added.
+			nodes = s.nodeRecycling[r1]
+			s.nodeRecycling = s.nodeRecycling[:r1]
+		} else {
+			// No nodes to recycle, make a new node slice. This happens when we scroll
+			// out a line that consisted of no screenlines.
+			nodes = make([]node, 0)
```

Comment:
The maxLines-disabled case preallocates the full width, so we should probably do the same here:

```suggestion
			nodes = make([]node, 0, s.cols)
```